### PR TITLE
Immediately logout of oauth.getcarina.com after oauth dance

### PR DESCRIPTION
**Pull request by carolynvs**
/cc @mdarby @rgbkrk @betatim Do you you think this would be an improvement over the current experience?

tldr: With this change, when a user logs out of a third party application and attempts to log back in, they aren't automatically logged-in as the previous user.

___

Here is the current flow:
_this is not currently deployed in production but will be this week_

1. Go to howtowhale and click login
2. Redirect to oauth.getcarina.com to enter username + password
3. Prompt to authorize howtowhale
4. Redirect back to howtowhale
5. Do fun stuff on howtowhale
6. Log out of howtowhale
7. Log back into howtowhale and without any prompts or confirmation, user is logged as the same
    user as before. 

There is no obvious way to sign in as a different user user or completely log out of Carina. In order to actually log out of Carina, user must go to oauth.getcarina.com and click logout.

This is how oauth works, that's not in question. With other services like GitHub, it's clear where to go when you want to log out of the main oauth application. However since you can't log out from oauth from getcarina.com (the first place everyone will look), it is confusing...

If we immediately log out the user from oauth.getcarina.com after the oauth dance, everything from the current flow is the same, except the last step. After logging out, when they try to log back into howtowhale again, they are prompted for their credentials.

Additionally, if the user goes to register a developer oauth application, they are able to sign in, and stay signed in as they navigate the oauth site. They are not logged out until they click logout, or do the oauth dance.

### CODE REVIEW

**Comment by rgbkrk** [comment]
We definitely have a difficult situation in that app.getcarina.com has it's own auth setup.

**Comment by mdarby** [comment]
I think it makes sense.

**Comment by carolynvs** [comment]
That's good enough for me! :shipit:

**Comment by betatim** [comment]
For what it is worth in `everware/everware` we use github oauth and provide a logout button on our pages. If you press it and return then you are asked again "Login with GitHub?" but if you visit a github page in the meanwhile you are still signed in. I was planning to investigate how that setup is different from whotowhale. They both use the `jupyter/oauthenticator` I think.

**Comment by carolynvs** [comment]
@betatim Yup! The way you are doing it with everware is the norm for oauth and what I would prefer to do in the long run.

I am trying this out mostly because we kind of hacked in support for oauth, whereas GitHub baked in oauth support directly into their main site. If you go to the official Carina site and logout, you aren't really logged out of oauth, which is contrary to what people expect based on their experiences with other oauth providers. If we ever move oauth directly into the main Carina site, then we would remove this "tweak".

I am going to deploy this week, update howtowhale accordingly and try it out for a bit. If it's really awful we can always come up with something else. :smile:
